lib/autokey/scripting/window.py

```diff
@@ -309,24 +314,28 @@ def get_window_list(self, filter_desktop=None):
         C{desktop} is the number of which desktop (sometimes called workspaces) the item appears upon.
 
         C{hostname} is the hostname of your computer (I'm not sure why this is included as it seems to be the same for everything I have seen)
-        
+
         C{title} is the title that you would usually see in your window manager of choice.
 
         @param filter_desktop: String, (usually 0-n) to filter the windows by. Any window not on the given desktop will not be returned.
         @return: C{[[hexid1, desktop1, hostname1, title1], [hexid2,desktop2,hostname2,title2], ...etc]}
         Returns C{[]} if no windows are found.
         """
-        windowList = self._run_wmctrl(["-l"])[1].split("\n")
-        output = []
-        for window in windowList:
-            info = window.split("  ")[1] #the remaining information
-            hexid = window.split("  ")[0] #the hexid of the window
-            desktop, hostname, window_title = info.split(" ", 2)
-            if filter_desktop is not None:
+        returncode, output = self._run_wmctrl(["-l"])
+        # regex = r"^(0x[0-9a-fA-F]{8})  (\d*) (.*?) (.*?)$"
+        regex = r"^(0x[0-9a-fA-F]{8})\s*(\d*)\s*(.*?)\s(.*?)$"
```

Comment:
Can you comment this regex?

Also, for regexes especially, tests are great for two reasons:
1. Easy enough to write, just have a bunch of examples and ensure it matches them and not others. Check `test_macro` for examples of regex tests.
2. Acts as a way to self-document exactly what the regex does.